### PR TITLE
Check for exit codes on each stage

### PR DESCRIPTION
**Pull request by jcagme**
@chcosta scheduled a build which failed for some reason (nothing was logged) and regardless of that the execution of the validation build continue to run. This change checks for exit codes different from 0, logs where the process failed and stops the execution since the current step normally depends on the previous.

### CODE REVIEW

File: eng/validate-sdk.ps1 [hunk 1]
```diff
@@ -8,11 +8,11 @@ $ErrorActionPreference = "Stop"
 . $PSScriptRoot\common\tools.ps1
 $LocalNugetConfigSourceName = "arcade-local"
 
-function Check-ExitCode ($exitCode)
+function Check-ExitCode([string]$stage)
```
Comment (mmitche) [inline eng/validate-sdk.ps1]: Should this maybe go into tools.ps1?

Comment (tmat) [reply eng/validate-sdk.ps1]: The message is still specific to this script.

Comment (jcagme) [reply eng/validate-sdk.ps1]: I did move it to tools.* and made the message more generic. Haven't tested it locally yet.

Comment (tmat) [reply eng/validate-sdk.ps1]: I'd actually not move it too tools. It's only used here and it's trivial to define.

**Comment by markwilkie** [comment]
@jcagme - next steps?

**Comment by jcagme** [comment]
If I remember correctly only validation was left. I'll check the current status if this on Tuesday

**Comment by chcosta** [comment]
I ran a local build earlier today on my machine (using valdatesdk), it failed to delete (or move) the artifacts folder which made it stop processing the script and then it ended claiming success.  I'll try to get a repro when I'm back in the office tomorrow.

**Comment by chcosta** [comment]
This is still hiding failures - https://dev.azure.com/dnceng/internal/_build/results?buildId=83821&view=logs&jobId=16ce49ee-8d0d-5a76-4958-88f78392b71f&taskId=206012c0-0eaa-56aa-8172-665f354177dc&lineStart=14651&lineEnd=14673&colStart=1&colEnd=1

**Comment by chcosta** [comment]
Is this being worked on?

Here are more instances

[3:09 PM] Cesar Soares Lucas
https://dnceng.visualstudio.com/internal/_build/results?buildId=83806
​
[3:09 PM] Cesar Soares Lucas
https://dnceng.visualstudio.com/internal/_build/results?buildId=84334

**Comment by chcosta** [comment]
I'm looking into fixing this

**Comment by chcosta** [comment]
This was fixed with PR #1980

**Pull request closed without merging**